serving-sdk/src/main/java/org/pytorch/serve/servingsdk/snapshot/SnapshotSerializer.java

```diff
@@ -0,0 +1,13 @@
+package org.pytorch.serve.servingsdk.snapshot;
+
+import java.io.IOException;
+import java.util.Properties;
+
+public interface SnapshotSerializer {
+
+    void saveSnapshot(Snapshot snapshot, final Properties prop) throws IOException;
```

Comment:
NIT: Should we throw a custom Snapshot exception instead? That way it will be less restrictive